jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +217,51 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repeatableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repeatableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repeatableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempTimeDelay = repeatableParts[1];
+				String tempDateTimeStr = repeatableParts[2];
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					if(DateTimeUtils.isPeriod(tempTimeDelay)) {
+						return new Date(System.currentTimeMillis() + Duration.parse(tempTimeDelay).toMillis() + Duration.parse(tempDateTimeStr).toMillis());
+					} else {
+						return new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + Duration.parse(tempDateTimeStr).toMillis());
+					}
+				} else {
+					if(DateTimeUtils.isPeriod(tempTimeDelay)) {
+						return new Date(System.currentTimeMillis() + Duration.parse(tempTimeDelay).toMillis() + DateTimeUtils.parseDateAsDuration(tempDateTimeStr));
+					} else {
+						return new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + DateTimeUtils.parseDateAsDuration(tempDateTimeStr));
```

Comment:
I think we misunderstood each other here :)
Right now you also handle `R/duration/duration` which I think doesn't make sense and it is not a part of ISO8601. You can have only 4 types according to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals):
1. `R/startDate/duration`
1. `R/duration/endDate`
1. `R/startDate/endDate`
1. `R/duration with context information` - this is parsed by `DateTimeUtils.parseISORepeatable(durationStr);` as if it was `R/now()/duration`

So 1. and 4. was handled correctly before in the first if - if `tempDateTimeStr` is Period, we take startDate + duration as a deadline.
The 2. and 3. (`tempDateTimeStr` is not Period) can be merged into one else branch since here we are only interested in deadline - so when the interval ends, so we can just take `tempDateTimeStr` and convert it into Date.

At least this is how I understand ISO8601 and our implementation of it in timers. @mswiderski wdyt?

Reply:
@MarianMacik correcting:
1 R/start/end 
2 R/start/duration
3 R/duration/end
4 R/duration

Reply:
Did I miss anything? I think I have enumerated every possibility... I think all of them make sense. In the first one there is end date as well so you can use it for deadline... Start may be specified just for users to have better context, ideally we could incorporate both start and end deadlines in one notation, but currently this is not the requirement...

Reply:
updated to remove the R / duration / duration option